EventBus.Kafka/KafkaEventPublisher.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Confluent.Kafka;
+using Finaps.EventBus.Core.Abstractions;
+using Finaps.EventBus.Kafka.Configuration;
+using Finaps.EventBus.Kafka.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.ObjectPool;
+using System;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace EventBus.Kafka
+{
+  public class KafkaEventPublisher : IEventPublisher
+  {
+    // private bool _disposed;
+    private readonly ILogger<KafkaEventPublisher> _logger;
+    private KafkaOptions _options;
+    private ObjectPool<IProducer<int, string>> _channelPool;
+    internal KafkaEventPublisher(
+      ILogger<KafkaEventPublisher> logger,
+      KafkaOptions options
+    )
+    {
+      _logger = logger;
+      _options = options;
+      _channelPool = CreateChannelPool();
+    }
+    
+    public void Publish(string message, string eventName, string messageId)
+    {
+      string topic = GetTopicFromKafkaMessage(message);
+      IProducer<int, string> channel = _channelPool.Get();
+      channel.Produce(topic, CreateMessage(message, eventName));
+      _channelPool.Return(channel);
+    }
+
+    public Task PublishAsync(string message, string eventName, string messageId)
+    {
+      string topic = GetTopicFromKafkaMessage(message);
+      IProducer<int, string> channel = _channelPool.Get();
+      channel.ProduceAsync(topic, CreateMessage(message, eventName));
+      _channelPool.Return(channel);
+      return Task.CompletedTask;
```

Comment:
I think the `ProduceAsync` should copy from the Azure implementation: [Link to code](https://github.com/Finaps/DotNetEventBus/blob/55055087ce68f3e5bfa3120b0fdc7d8a58ae7dc2/EventBus.AzureServiceBus/AzureServiceBusEventPublisher.cs#L42)